### ISSUE TITLE
Imbuing feedback (as it currently is at the moment)

### ISSUE THREAD

**Issue by Cruuk** [Tslat/Advent-Of-Ascension#4092]
Survival world context: 24+ hours in this world, close-ish to finishing it. Level 90 Hauling, level 50 Extraction, 75 Dexterity, 36 Innervation, **30 Imbuing**. Got a Utopian helm early on from a Treasure box. Tried to focus on training Imbuing by generally sticking to 1 area and periodically running around the area in big circles hunting for Pixons. Also, this is coming from someone who already has an idea of the spawn mechanics and tried my best to utilize that knowledge.

-----------------------------

Currently, it has a nice foundation but has some fairly significant balancing issues. I also want to see Imbuing have some dim progression stuff in it, but I'll save that for another git post.

-----------------------------

**__Issues__**
The main overarching issues is searching for the Pixons, getting exp for the skill after a certain point, power stone rarity, and benefitting from the skill in general.

Is the Pixon spawnrate fine? It's hard to say, but maybe it is? I think the issue is moreso the spawn radius being so wide. It got annoying running in giant circles, only getting 1 Pixon ~50% of the time in run-arounds that took 8 minutes or so. Because it's timer-based and can vary, I never knew if I'd go outward at the wrong time and have Pixons spawn at even greater distances I never checked. There were sparse forests, but it was otherwise a big open area that I always made sure to watch for Pixons in.

I only saw Pixons in the outer layer of places I'd spend time in (I.E. the outward area of the biome my base was in 95% of the time). When I traveled long distances to specific structures/biomes, I never saw any while traveling, only after staying in 1 area. The couple exceptions to Pixons being near my base were me surfacing after being underground for a while (there was only like 1 Pixon)

There's this visual bug with rain where the rain will light up in the general vicinity of a Pixon (doesn't always happen). I used this to my advantage, because I REALLY did not want to miss any of these damn things.

The exp rate is good - for the first ~24 levels or so. After that, it slowed to a crawl. You mentioned that Pixons are supposed to be the main exp source early on, then imbuing stuff becomes the main exp source later. Unfortunately, I have barely been able to use Power stones due to them being very rare.

I did get 1 from at least ~50% of the Pixons I've harvested, but even then, it never felt I like could use them to really benefit me due to their practical infrequency.

In general, despite my best efforts to focus on Imbuing, it hasn't really felt like it's benefitted me at all. The only exception is using Ambient power stones to put lure/luck of the sea on Hauling rods and combine them in an anvil LOL. That's the only times I've really imbued.

This is mainly due to the combination of power stone rarity and Pixon tiers - both being at odds with my Imbuing level.

For the power stone rarity, I feel like you have either make the Pixon OR Power stone rare. Having both be rare may cause issues.

In terms of the Pixon tiers I've seen, it's been mostly Ambient/Glistening, a couple blooming, 1 shining, and 1 radiant. I am stuck in limbo with the level I'm at - the enchants Ambient/glistening power stones give me are too weak unless I do anvil stuff with multiples of the same weapon. I did get 1 shining power stone, but I can't use it besides furnace fuel - the closest enchant is level 68.

Though I guess enchanting by enchant table didn't help the viability of the lower tier power stones. Even if there were a bunch of Imbuing-exclusive enchants, getting the right power stone you can actually use will be quite a hassle.

-----------

**__Solutions__**
For Pixon spawning, I do think the radius should be like 50% higher than the roaming traders instead of 100% so I don't have to run around as much. If possible, also increase the radius you can hear their sound at so I don't have to use buggy rain to make sure I don't miss them.

For exp stuff, I do think enchanting at enchant table should give Imbuing exp. It shouldn't be a solid source of exp, just a thing to help earlygame and be a fallback if you're lacking imbuing resources like power stones.

Decreasing power stone rarity via increasing their drop chance from Pixons. I want them to be a rare item, but mainly the higher tiers (Gleaming --> Radiant), I don't think it would hurt for Ambient, Glistening, and Blooming to be more common so I can actually use them in ways that matter. 

I'm gonna repeat this point in the other git post, but I do think it would be neat if the tier of a dim modified what Pixons can spawn in it - Ambient can't spawn in T1 dims, Ambient and Glistening can't spawn in T2, etc with cutting out the lowest ones. It would still keep the absolute highest ones at a lower chance, but it would be nice to have a method of not getting stuck with lower tier pixons that can't benefit you anymore - Ambient pixon already gives tiny exp amounts at level 20.

I do think imbuing in general should be a direct upgrade to enchant table considering the resource cost/Imbuing level requirement. Enchantment table would still have the upsides of random enchants only costing xp/Lapis. The aspect Foci are fine by the way, it's mainly power stones (especially the lower tier ones) holding the skill back

**Comment by Tslat** [comment]
Yeah I had a feeling this might be the case

Honestly I've been wondering whether it's worth getting rid of energy stones entirely and making power stones much more common but giving them more uses

I'm also still looking for more ability ideas for imbuing in general, but it's proving difficult

**Comment by Cruuk** [comment]
> Yeah I had a feeling this might be the case
> 
> Honestly I've been wondering whether it's worth getting rid of energy stones entirely and making power stones much more common but giving them more uses
> 
> I'm also still looking for more ability ideas for imbuing in general, but it's proving difficult

I don't think it hurts for Pixons having a guaranteed drop with Energy stones. Maybe you could make it so you can convert like 12-16 energy stones into 1 power stone (could be an early Imbuing ability). Though if you can't find more uses for energy stones, I guess they would be redundant at that point

**Comment by Seishan123** [comment]
Don't know if you can make this possible because of the way skills gain experience, but perhaps making bowls support the fortune enchantment to increase the amount of stones gained from pixons (and subsequently experience) could help with this issue. Even then... I remember years ago spending literal days running around dimensions (or flying with Elytra) farming pixons for a level or two.
The only other idea I have is maybe building a specific structure around the Infusion Table to increase odds of getting Power Stones on use. I've seen this done in other mods in the past, especially magic mods... Perhaps the schematics/special blocks or items for them can be found in certain structures across dimensions (if only to encourage some more exploration and keep it from being accessible too early).
